### PR TITLE
Feature/update importer

### PR DESCRIPTION
**Pull request by rtisma**
- added GET url based searching, aswell as POST based searching of analysis objects
- very primitive, can only search by studyId and donorId/specimenId/sampleId/fileId, and its key-value (no donorIds)

### CODE REVIEW

File: song-importer/src/main/java/org/icgc/dcc/song/importer/Importer.java [hunk 1]
```diff
@@ -163,11 +163,11 @@ private void processAnalysis(DataContainer dataContainer){
       analysisRepository.createVariantCall(x.getExperiment());
     });
 
-    log.info("Converting FileSets...");
-    val fileSets = FILE_SET_CONVERTER.convertFileSets(dataContainer.getPortalFileMetadataList());
+//    log.info("Converting FileSets...");
```
Comment (andricDu) [inline song-importer/src/main/java/org/icgc/dcc/song/importer/Importer.java]: remove commented code.

File: song-importer/src/main/java/org/icgc/dcc/song/importer/convert/AnalysisConverter.java [hunk 2]
```diff
@@ -126,15 +125,19 @@ private static SequencingReadAnalysis buildSequencingReadAnalysis(SeqReadAggrega
   }
 
   private static void updateAnalysis(Analysis analysis, SeqReadAggregate seqReadAggregate){
-    analysis.setAnalysisId(seqReadAggregate.getAnalysisId());
+    val analysisId = seqReadAggregate.getAnalysisId();
```
Comment (andricDu) [inline song-importer/src/main/java/org/icgc/dcc/song/importer/convert/AnalysisConverter.java]: nothing to change here at the moment, but using methods for updating objects through side effect is pretty bad. The void return means you can't chain calls or use properly in a stream.

File: song-server/src/main/java/org/icgc/dcc/song/server/model/analysis/AnalysisSearchRequest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    return procOpt(this.sampleId);
+  }
+  public String getSpecimenId(){
+    return procOpt(this.specimenId);
+  }
+  public String getFileId(){
+    return procOpt(this.fileId);
+  }
+
+  private static String procOpt(String opt){
```
Comment (andricDu) [inline song-server/src/main/java/org/icgc/dcc/song/server/model/analysis/AnalysisSearchRequest.java]: what is procOpt?

File: song-server/src/main/java/org/icgc/dcc/song/server/service/AnalysisService.java [hunk 4]
```diff
@@ -184,4 +187,27 @@ public Analysis read(String id) {
...
   }
+
+  /**
+   * Adds all child entities for each analysis
+   * This method should be watched in case performance becomes a problem.
+   * @param analysisList list of Analysis to be updated
+   * @return returns a List of analysis with the child entities.
+   */
+  private List<Analysis> processAnalysisList(List<Analysis>  analysisList){
+    analysisList.forEach(a -> {
```
Comment (andricDu) [inline song-server/src/main/java/org/icgc/dcc/song/server/service/AnalysisService.java]: getting large for a lambda. Factor out into method and use a method reference.

File: song-server/src/main/resources/queries/analysis/searchAnalysis.sql [hunk 5]
```diff
@@ -0,0 +1,18 @@
+SELECT DISTINCT
```
Comment (andricDu) [inline song-server/src/main/resources/queries/analysis/searchAnalysis.sql]: cool!

**Comment by andricDu** [comment]
This is a good size for a PR 👍

File: song-importer/src/main/java/org/icgc/dcc/song/importer/convert/AnalysisConverter.java (hunk 2 above)
Comment (rtisma) [inline song-importer/src/main/java/org/icgc/dcc/song/importer/convert/AnalysisConverter.java]: yea this smelt bad to begin with, my bad.

**Pull request merged**